### ISSUE TITLE
SelfAssignment false positive with casting a float variable to int

### ISSUE THREAD

**Issue by DevCharly** [google/error-prone#4437]
### Description of the problem:

The SelfAssignment checker complains about following:

```java
float width = 123.456f;

width = (int) width;

// value of width is now 123
```

### What version of Error Prone are you using?

2.28.0

**Comment by cushon** [comment]
Some previous discussion in https://github.com/google/error-prone/issues/2196
